MSAL/test/unit/MSALInteractiveRequestTests.m

```diff
@@ -707,4 +707,150 @@ - (void)testInteractiveRequestFlowWithUserNotMatch
+    parameters.correlationId = correlationId;
+    parameters.tokenCache = [MSALTestTokenCache createTestAccessor];
+    
+    [MSALTestSwizzle classMethod:@selector(randomUrlSafeStringOfSize:)
+                           class:[NSString class]
+                           block:(id)^(id obj, NSUInteger size)
+     {
+         (void)obj;
+         (void)size;
+         return @"randomValue";
+     }];
+    
+    MSALPkce *pkce = [MSALPkce new];
+    
+    __block MSALInteractiveRequest *request =
+    [[MSALInteractiveRequest alloc] initWithParameters:parameters
+                                  extraScopesToConsent:@[@"fakescope3"]
+                                              behavior:MSALForceConsent
+                                                 error:&error];
+    
+    XCTAssertNotNil(request);
+    XCTAssertNil(error);
+    
+    // Setting MSALAuthority ahead of time short-circuits authority validation for this test
+    request.authority = [MSALTestAuthority AADAuthority:parameters.unvalidatedAuthority];
+    
+    // Swizzle out the main entry point for WebUI, WebUI is tested in its own component tests
+    [MSALTestSwizzle classMethod:@selector(startWebUIWithURL:context:completionBlock:)
+                           class:[MSALWebUI class]
+                           block:(id)^(id obj, NSURL *url, id<MSALRequestContext>context, MSALWebUICompletionBlock completionBlock)
+     {
+         (void)obj;
+         (void)context;
+         
+         XCTAssertNotNil(url);
+         XCTAssertEqualObjects(url.scheme, @"https");
+         XCTAssertEqualObjects(url.msalHostWithPort, @"login.microsoftonline.com");
+         XCTAssertEqualObjects(url.path, @"/common/oauth2/v2.0/authorize");
+         
+         NSDictionary *msalId = [MSALLogger msalId];
+         NSDictionary *expectedQPs =
+         @{
+           @"x-client-Ver" : MSAL_VERSION_NSSTRING,
+#if TARGET_OS_IPHONE
+           @"x-client-SKU" : @"MSAL.iOS",
+           @"x-client-DM" : msalId[@"x-client-DM"],
+#else
+           @"x-client-SKU" : @"MSAL.OSX",
+#endif
+           @"x-client-OS" : msalId[@"x-client-OS"],
+           @"x-client-CPU" : msalId[@"x-client-CPU"],
+           @"return-client-request-id" : correlationId.UUIDString,
+           @"state" : request.state,
+           @"prompt" : @"consent",
+           @"login_hint" : @"fakeuser@contoso.com",
+           @"client_id" : UNIT_TEST_CLIENT_ID,
+           @"scope" : @"fakescope1 fakescope2 fakescope3 openid profile offline_access",
+           @"eqp1" : @"val1",
+           @"eqp2" : @"val2",
+           @"redirect_uri" : UNIT_TEST_DEFAULT_REDIRECT_URI,
+           @"response_type" : @"code",
+           @"code_challenge": pkce.codeChallenge,
+           @"code_challenge_method" : @"S256",
+           @"uid" : @"true",
+           @"slice" : @"testslice"
+           };
+         NSDictionary *QPs = [NSDictionary msalURLFormDecode:url.query];
+         XCTAssertTrue([expectedQPs compareToActual:QPs]);
```

Comment:
yes, I was also thinking that tests in here are very repetitive and can probably be improved.